test/unit/buffer_spec.lua

```diff
@@ -212,92 +213,164 @@ describe('buffer functions', function()
 
   describe('build_stl_str_hl', function()
 
-    local output_buffer = to_cstr(string.rep(" ", 100))
-
-    local build_stl_str_hl = function(pat)
-      return buffer.build_stl_str_hl(globals.curwin,
-                                     output_buffer,
-                                     100,
-                                     to_cstr(pat),
-                                     false,
-                                     32,
-                                     80,
-                                     NULL,
-                                     NULL)
-    end
-
-    it('should copy plain text', function()
-      local width = build_stl_str_hl("this is a test")
-
-      eq(14, width)
-      eq("this is a test", helpers.ffi.string(output_buffer, width))
-
+    before_each(function()
+      buffer_width = 100
```

Comment:
Why this is in `before_each` at all, not to mention it is not local? Even if you must define it here for some reason, you should declare it local in the outer scope (i.e. just before `before_each`).